src/utils/havoc.js

```diff
@@ -322,10 +324,6 @@ class ObjectValueHavocingVisitor {
       return;
     }
 
-    // TODO: if this object is frozen there is no need to havoc it.
-    // TODO: if the object is not extensible, just havoc the known properties
-    // TODO: if a property is readonly and not configurable, leave it alone
```

Comment:
The last two todos are not addressed by this PR. They should remain.

Reply:
The last TODO is kind of pointless with the current implementation of havoc, which I did not fully understand when I wrote the TODO. The second to last TODO no longer seems very valuable to me since it only retains knowledge about which properties are not in the havoced object. It does not seem likely that such knowledge will help anything very much, so doing work to make it happen does not seem a good investment.